Upstream-Status: Pending
diff --git a/fs/fs.c b/fs/fs.c
index acf465b..8ec6ff6 100644
--- a/fs/fs.c
+++ b/fs/fs.c
@@ -784,7 +784,7 @@ int do_load(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[],
 		pos = 0;
 
 	time = get_timer(0);
-	ret = _fs_read(filename, addr, pos, bytes, 1, &len_read);
+	ret = _fs_read(filename, addr, pos, bytes, 0, &len_read);
 	time = get_timer(time);
 	if (ret < 0) {
 		log_err("Failed to load '%s'\n", filename);
